src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Registry.hs

```diff
@@ -72,8 +72,9 @@ removeStructure fs (FoundRegistry byName byLoc) =
   -- Swarm.Game.State.removeRobotFromLocationMap
   tidyDelete = NEM.nonEmptyMap . NEM.delete upperLeft
 
-addFound :: FoundStructure b a -> FoundRegistry b a -> FoundRegistry b a
-addFound fs@(FoundStructure swg loc) (FoundRegistry byName byLoc) =
+addFound :: Maybe (FoundStructure b a) -> FoundRegistry b a -> FoundRegistry b a
+addFound Nothing x = x
```

Comment:
Might be worth putting `addFound` back to the way it was (without the `Maybe`), then we can express the new `addFound`  as `maybe id addFound` (which could be inlined into the use site, or defined as a separate function).